i18n/src/main/kotlin/com/kamelia/sprinkler/i18n/TranslatorBuilder.kt

```diff
@@ -50,59 +52,65 @@ class TranslatorBuilder @PackagePrivate internal constructor(
     }
 
+    /**
+     * Adds a path to the builder. If the path is a directory, all files in it will be loaded. If the path is a file, it
+     * will be loaded.
+     *
+     * This method will throw an [IllegalArgumentException] if the path is already added.
+     *
+     * This method is a shorthand for [addPath] with a [I18nFileParser.fromString] parser.
+     *
+     * @param path the path to load
+     * @param mapper the mapper to use to load the file
+     * @return this builder
+     * @throws IllegalArgumentException if the path is already added
+     * @see I18nFileParser.fromString
+     */
+    fun addPath(path: Path, mapper: (String) -> Map<String, Any>): TranslatorBuilder =
+        addPath(path, I18nFileParser.fromString { mapper(it) })
+
     /**
      * Adds a file to the builder. If the file is a directory, all files in it will be loaded. If the file is a file, it
      * will be loaded.
      *
-     * Content will be converted to a map using the given [parser], and the locale of the file will be determined using
-     * the given [localeMapper]. By default, the file name is parsed using [Locale.forLanguageTag].
+     * This method will throw an [IllegalArgumentException] if the path is already added.
+     *
+     * @param file the file to load
+     * @param parser the parser to use to load the file
+     * @return this builder
+     * @throws IllegalArgumentException if the file is already added
+     */
+    fun addFile(file: File, parser: I18nFileParser): TranslatorBuilder = addPath(file.toPath(), parser)
+
+    /**
+     * Adds a file to the builder. If the file is a directory, all files in it will be loaded. If the file is a file, it
+     * will be loaded.
      *
      * This method will throw an [IllegalArgumentException] if the path is already added.
      *
-     * **NOTE**: If the default implementation of [localeMapper] is used, any invalid locale parsed during [build] will
-     * lead to an [IllegalStateException] being thrown.
+     * This method is a shorthand for [addFile] with a [I18nFileParser.fromString] parser.
      *
      * @param file the file to load
-     * @param localeMapper the mapper to use to map the file name to a locale (by default, the file name is parsed using
-     * [Locale.forLanguageTag])
-     * @param parser the parser to use to load the file
+     * @param mapper the mapper to use to load the file
      * @return this builder
      * @throws IllegalArgumentException if the file is already added
+     * @see I18nFileParser.fromString
      */
-    @JvmOverloads
-    fun addFile(
-        file: File,
-        localeMapper: (String) -> Locale = ::parseLocale,
-        parser: I18nFileParser,
-    ): TranslatorBuilder = apply {
-        addPath(file.toPath(), localeMapper, parser)
-    }
+    fun addFile(file: File, mapper: (String) -> Map<String, Any>): TranslatorBuilder =
+        addFile(file, I18nFileParser.fromString { mapper(it) })
```

Comment:
skip location parser and pass function ref

```suggestion
        addFile(file, I18nFileParser.fromString(mapper = mapper))
```